### ISSUE TITLE
How long did TSM take to train from scratch on Kinectics ?

### ISSUE THREAD

**Issue by malarsaravanan1991** [mit-han-lab/temporal-shift-module#28]
Hey Hi,

Thank you for your work.
 I am trying to train TSM **online version on kinectics with resnet 50** and it has been two days and it has not passed two epocs. 

How long did it take to train TSM network from scratch online version for both for resnet 50 and mobilenetv2? I just wanted to make sure if i am in the right path.

**Comment by tonylins** [comment]
Hi, if you train the model on 8 GPUs, it should finish in roughly 2 days.

**Comment by malarsaravanan1991** [comment]
Thanks

**Comment by just-justin144** [comment]
Hello, If possible could you specify which GPU's where used? And where you bottlenecked by IO for the video data?

**Comment by tonylins** [comment]
Hi, to be more specific, it takes about 49 hours to train a ResNet-50 TSM on Kinetics using 6 V100 GPUs. So it is about 1.5 days to train with 8 V100 GPUs.

Regarding data loading, if you are training an 8-frame model, data loading usually does not delay training. But it might be slow to load data when taking 16 or more frames as input, depending on the hardware.

**Comment by just-justin144** [comment]
Thank you very much!

**Comment by malarsaravanan1991** [comment]
One last quick question. We are trying to train mobilenet V2 from scratch the online version on kinetics.Can you share us the hyper parameter that you had used in training mobilenetv2 online version on kinetics dataset?
parameters like lr,epochs,weight decay,batch size,num of workers?

Also confirm if  the change below is correct for the online training script(i just modified the shift part of the code such that the shift is happening unidirectional in channel dimension),or do we have to make any more changes ?

> Temporal shift implementation of online version

            out = torch.zeros_like(x)
            **out[:, :1, :fold] = x[:, -1:, :fold]  # shifts one way**
            out[:, 1:, fold: 2 * fold] = x[:, :-1, fold: 2 * fold]  # shift right
            out[:, :, 2 * fold:] = x[:, :, 2 * fold:]  # not shift

**Comment by ucasiggcas** [comment]
hi,dear
could you please help me ?
https://github.com/mit-han-lab/temporal-shift-module/issues/43
https://github.com/mit-han-lab/temporal-shift-module/issues/42
https://github.com/mit-han-lab/temporal-shift-module/issues/41
thx
